### PR TITLE
docs: add docs for member profile endpoints

### PR DESCRIPTION
**Pull request by pyphilia**
ref #1528

### CODE REVIEW

File: src/services/member/plugins/profile/controller.ts [hunk 1]
```diff
@@ -68,7 +73,7 @@ const plugin: FastifyPluginAsyncTypebox = async (fastify) => {
   fastify.patch(
     '/',
     {
-      schema: updateMemberProfile,
+      schema: updateOwnMemberProfile,
```
Comment (spaenleh) [inline src/services/member/plugins/profile/controller.ts:76]:
```suggestion
      schema: updateOwnProfile,
```

File: src/services/member/plugins/profile/schemas.ts [hunk 2]
```diff
@@ -6,68 +6,85 @@ import { FastifySchema } from 'fastify';
 import { customType, registerSchemaAsRef } from '../../../../plugins/typebox';
 import { errorSchemaRef } from '../../../../schemas/global';
 
-const inputPublicProfileMemberSchemaRef = registerSchemaAsRef(
-  'publicProfile',
-  'Public Profile',
+const inputPublicProfileMemberSchemaRef = customType.StrictObject({
```
Comment (spaenleh) [inline src/services/member/plugins/profile/schemas.ts:9]: This is not a ref anymore.
```suggestion
const inputPublicProfileMemberSchema = customType.StrictObject({
```

File: src/services/member/plugins/profile/schemas.ts [hunk 3]
```diff
@@ -6,68 +6,85 @@ import { FastifySchema } from 'fastify';
...
     [StatusCodes.OK]: profileMemberSchemaRef,
     // Status NO CONTENT is used instead of NOT FOUND, so it doesn't trigger an error in the Frontend
     [StatusCodes.NO_CONTENT]: Type.Null(),
     [StatusCodes.UNAUTHORIZED]: errorSchemaRef,
+    '4xx': errorSchemaRef,
   },
 } as const satisfies FastifySchema;
 
 export const getOwnProfile = {
+  operationId: 'getOwnMemberProfile',
```
Comment (spaenleh) [inline src/services/member/plugins/profile/schemas.ts:64]:
```suggestion
  operationId: 'getOwnProfile',
```

File: src/services/member/plugins/profile/schemas.ts [hunk 4]
```diff
@@ -6,68 +6,85 @@ import { FastifySchema } from 'fastify';
...
     [StatusCodes.OK]: profileMemberSchemaRef,
     // Status NO CONTENT is used instead of NOT FOUND, so it doesn't trigger an error in the Frontend
     [StatusCodes.NO_CONTENT]: Type.Null(),
     [StatusCodes.UNAUTHORIZED]: errorSchemaRef,
+    '4xx': errorSchemaRef,
   },
 } as const satisfies FastifySchema;
 
-export const updateMemberProfile = {
+export const updateOwnMemberProfile = {
```
Comment (spaenleh) [inline src/services/member/plugins/profile/schemas.ts:78]:
```suggestion
export const updateOwnProfile = {
```

File: src/services/member/plugins/profile/schemas.ts [hunk 5]
```diff
@@ -6,68 +6,85 @@ import { FastifySchema } from 'fastify';
...
-    facebookID: Type.String(),
-    linkedinID: Type.String(),
-    twitterID: Type.String(),
-    visibility: Type.Boolean(),
-    createdAt: customType.DateTime(),
-    updatedAt: customType.DateTime(),
-  }),
-);
+export const createOwnProfile = {
+  operationId: 'createOwnMemberProfile',
```
Comment (spaenleh) [inline src/services/member/plugins/profile/schemas.ts:34]:
```suggestion
  operationId: 'createOwnProfile',
```

**Review by spaenleh — CHANGES_REQUESTED** [review]
Nice, I would adapt the naming a bit.

**Review by spaenleh — APPROVED** [review]
🌻

**Review by ReidyT — APPROVED** [review]

**Pull request merged**